### PR TITLE
add support for GIT download on unix

### PR DESCRIPTION
**Pull request by stiliev**

### CODE REVIEW

File: src/Misc/externals.sh [hunk 1]
```diff
@@ -42,11 +57,36 @@ get_abs_path() {
...
 DOWNLOAD_DIR=$(get_abs_path `dirname $0`/../../_downloads)
 
-PLATFORM_NAME=`uname`
-PLATFORM="windows"
-if [[ ("$PLATFORM_NAME" == "Linux") || ("$PLATFORM_NAME" == "Darwin") ]]; then
-   PLATFORM=`echo "${PLATFORM_NAME}" | awk '{print tolower($0)}'`
-fi
+set -o pipefail
+
+get_current_os_name() {
```
Comment (TingluoHuang) [inline src/Misc/externals.sh]: function getCurrentOSName()

File: src/Misc/externals.sh [hunk 2]
```diff
@@ -42,11 +57,36 @@ get_abs_path() {
 LAYOUT_DIR=$(get_abs_path `dirname $0`/../../_layout)
 DOWNLOAD_DIR=$(get_abs_path `dirname $0`/../../_downloads)
 
-PLATFORM_NAME=`uname`
-PLATFORM="windows"
-if [[ ("$PLATFORM_NAME" == "Linux") || ("$PLATFORM_NAME" == "Darwin") ]]; then
-   PLATFORM=`echo "${PLATFORM_NAME}" | awk '{print tolower($0)}'`
-fi
+set -o pipefail
```
Comment (TingluoHuang) [inline src/Misc/externals.sh]: don't forget to remove.

**Comment by TingluoHuang** [comment]
Looks good. :D

File: src/Misc/externals.sh (hunk 1 above)
Comment (stilkolev) [inline src/Misc/externals.sh]: will fix

Comment (stiliev) [inline src/Misc/externals.sh]: will fix

File: src/Misc/externals.sh (hunk 2 above)
Comment (stiliev) [inline src/Misc/externals.sh]: will fix

**Pull request merged**